### PR TITLE
Docs: Revised structure of documentation

### PR DESCRIPTION
**Pull request by marcogario**
* Added Full API reference
* Updated Hello World example
* Changed info@pysmt.org to Google group

### CODE REVIEW

File: pysmt/shortcuts.py [hunk 1]
```diff
@@ -551,6 +553,14 @@ def sequence_interpolant(formulas, solver_name=None, logic=None):
                                             solver_name=solver_name,
                                             logic=logic)
 
+def read(fname):
+    """Reads the SMT formula from the given file."""
+    raise NotImplementedError
```
Comment (marcogario) [inline pysmt/shortcuts.py]: This should be implemented.

Comment (marcogario) [inline pysmt/shortcuts.py]: See #295

**Comment by marcogario** [comment]
I activated the build for this branch on readthedocs, it can accessed at:

http://pysmt.readthedocs.io/en/i276-documentation/

File: docs/conf.py [hunk 2]
```diff
@@ -49,17 +49,17 @@
...
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = '0.2'
+version = '0.5'
 # The full version, including alpha/beta/rc tags.
-release = '0.2.2'
+release = '0.5.1'
```
Comment (mikand) [inline docs/conf.py]: This has been updated in master, maybe we should rebase??

File: pysmt/solvers/solver.py [hunk 3]
```diff
@@ -334,7 +333,16 @@ def reset_assertions(self):
...
+        """Assert the formula in the solver.
+
+        This must return the asserted formula (as an FNode) exactly as
+        it was asserted in the solver, thus accounting for rewritings,
+        simplifications, etc.
+
+        :returns: The asserted formula
+        :rtype: :py:class:
+        """
+        raise NotImplementbedError
```
Comment (marcogario) [inline pysmt/solvers/solver.py]: typo

**Pull request merged**